src/pages/RegisterStudents.vue

```diff
@@ -317,102 +318,114 @@ function submitStudents(rawJson) {
+        if (!_isEmpty(id)) {
+          _set(sendObject, 'userData.class', id)
+        } else {
+          addErrorUser(user, `Error: Class '${uClass}' is invalid.`)
+          if(processedUsers >= totalUsers){
+            activeSubmit.value = false;
+          }
+          return;
         }
-        return;
       }
-    }
 
-    // If group is a given column, check if the name is
-    //   associated with a valid id. If so, add the id to
-    //   the sendObject. If not, reject user
-    if (group) {
-      const id = getGroupId(group);
-      if (!_isEmpty(id)) {
-        _set(sendObject, 'userData.group', id)
-      } else {
-        addErrorUser(user, `Error: Group '${group}' is invalid.`)
-        if(processedUsers >= totalUsers){
-          activeSubmit.value = false;
+      // If group is a given column, check if the name is
+      //   associated with a valid id. If so, add the id to
+      //   the sendObject. If not, reject user
+      if (group) {
+        const id = getGroupId(group);
+        if (!_isEmpty(id)) {
+          _set(sendObject, 'userData.group', id)
+        } else {
+          addErrorUser(user, `Error: Group '${group}' is invalid.`)
+          if(processedUsers >= totalUsers){
+            activeSubmit.value = false;
+          }
+          return;
         }
-        return;
       }
-    }
 
-    authStore.registerWithEmailAndPassword(sendObject).then(() => {
-      toast.add({ severity: 'success', summary: 'User Creation Success', detail: `${sendObject.email} was sucessfully created.`, life: 9000 });
-      processedUsers = processedUsers + 1;
-      if(processedUsers >= totalUsers){
-        activeSubmit.value = false;
-        if(errorUsers.value.length === 0) {
-          // Processing is finished, and there are no error users.
-          router.push({ name: "Home" })
+      authStore.registerWithEmailAndPassword(sendObject).then(() => {
+        toast.add({ severity: 'success', summary: 'User Creation Success', detail: `${sendObject.email} was sucessfully created.`, life: 9000 });
+        processedUsers = processedUsers + 1;
+        if(processedUsers >= totalUsers){
+          activeSubmit.value = false;
+          if(errorUsers.value.length === 0) {
+            // Processing is finished, and there are no error users.
+            router.push({ name: "Home" })
+          }
         }
-      }
-    }).catch((e) => {
-      toast.add({ severity: 'error', summary: 'User Creation Failed', detail: 'Please see error table below.', life: 3000 });
-      addErrorUser(user, e)
-      if(processedUsers >= totalUsers){
-        activeSubmit.value = false;
-      }
+      }).catch((e) => {
+        toast.add({ severity: 'error', summary: 'User Creation Failed', detail: 'Please see error table below.', life: 3000 });
+        addErrorUser(user, e)
+        if(processedUsers >= totalUsers){
+          activeSubmit.value = false;
+        }
+      })
     })
-  })
+    await delay(1000)
```

Comment:
I suggest padding to reduce possible overlap due to cloud function queueing instability.
```suggestion
    await delay(1250)
```